algorithms/CPlusPlus/Maths/armstrong.cpp

```diff
@@ -0,0 +1,19 @@
+#include <iostream>
+using namespace std;
+int main() {
+    int num, i=0,r,sum=0,s;
+    //cout <<"enter a number to check it is Armstrong no. or not ";
+    cin>>num; 
+    s=num;
+    while (i<num)
+    {
+        r=num%10;
+        sum =sum+(r*r*r);
```

Comment:
As @Ashborn-SM already pointed out, the program doesn't work for numbers with less or more digits than a three digit number.

By definition, An [Armstrong number ( or Narcissistic_number )](https://en.wikipedia.org/wiki/Narcissistic_number) is a number that is the sum of its own digits each raised to the power of the **number of digits**.